src/api/libcellml/units.h

```diff
@@ -0,0 +1,182 @@
+const std::string STANDARD_UNIT_OHM = "ohm"; /**< Derived SI unit ohm. */
+const std::string STANDARD_UNIT_PASCAL = "pascal"; /**< Derived SI unit pascal. */
+const std::string STANDARD_UNIT_RADIAN = "radian"; /**< Derived SI unit radian. */
+const std::string STANDARD_UNIT_SECOND = "second"; /**< Base SI unit second. */
+const std::string STANDARD_UNIT_SIEMENS = "siemens"; /**< Derived SI unit siemens. */
+const std::string STANDARD_UNIT_SIEVERT = "sievert"; /**< Derived SI unit sievert. */
+const std::string STANDARD_UNIT_STERADIAN = "steradian"; /**< Derived SI unit steradian. */
+const std::string STANDARD_UNIT_TESLA = "tesla"; /**< Derived SI unit tesla. */
+const std::string STANDARD_UNIT_VOLT = "volt"; /**< Derived SI unit volt. */
+const std::string STANDARD_UNIT_WATT = "watt"; /**< Derived SI unit watt. */
+const std::string STANDARD_UNIT_WEBER = "weber"; /**< Derived SI unit weber. */
+
+/**
+ * @brief The Units class.
+ * Class for Units.
+ */
+class LIBCELLML_EXPORT Units: public NamedEntity
+{
+public:
+    Units(); /**< Constructor */
+    ~Units(); /**< Destructor */
+    Units(const Units &rhs); /**< Copy constructor */
+    Units(Units &&rhs); /**< Move constructor */
+    Units& operator=(Units n); /**< Assignment operator */
+
+    /**
+     * @brief Test to determine if Units is a base unit.
+     * Test to determine if Units is a base unit, return True if it is
+     * a base unit and False otherwise.
+     * @return True if Units is a base unit, False otherwise.
+     */
+    bool isBaseUnit() const;
+
+    /**
+     * @brief Set whether this unit is a base unit or not.
+     * Set the base unit flag to signify whether this Units is a base unit
+     * or not.  This flag cannot be set if this Units is a compound unit.
+     * @param state boolean value to set the base unit flag to.
+     */
+    void setBaseUnit(bool state=true);
+
+    /**
+     * @brief Add a unit to this Units.
+     * Add a unit as a child of this Units.  This method takes optional arguments
+     * multiplier and offset.
+     * @param units The name of the unit to add.
+     * @param prefix The prefix for the unit, one of PREFIXES.
+     * @param exponent The exponent.
+     * @param multiplier The multiplier.
+     * @param offset The offset.
+     */
+    void addUnit(const std::string &units, PREFIXES prefix, double exponent,
+                 double multiplier=1.0, double offset=0.0);
+
+    /**
+     * @brief Add a unit to this Units.
+     * Add a unit as a child of this Units.  This method takes optional arguments
+     * multiplier and offset.
+     *
+     * @overload
+     * @param units The name of the unit to add.
+     * @param prefix The prefix for the unit expressed as an integer.
+     * @param exponent The exponent.
+     * @param multiplier The multiplier.
+     * @param offset The offset.
+     */
+    void addUnit(const std::string &units, int prefix, double exponent,
```

Comment:
could you give `exponent` a default value of `1` and then you wouldn't need the following two methods?